lib/sharing/ConsumeSharedPlugin.js

```diff
@@ -252,6 +256,62 @@ class ConsumeSharedPlugin {
 								requiredVersion
 							}
 						);
+						// Check if exclusion criteria is provided
+						if (config.exclusionCriteria && config.exclusionCriteria.version) {
+							// Without a resolved import, we cannot proceed with exclusion check
+							if (!importResolved) {
+								return consumedModule;
+							}
```

Comment:
Here, we check if `importResolved` can be mapped out. If there is no resolved import path for a request, we cannot decide whether to exclude it from sharing, so by default `include` it.